storage/innobase/os/os0file.cc

```diff
@@ -4769,6 +4796,75 @@ readahead requests. */
 #endif /* __WIN__ */
 }
 
+/** Submit buffered AIO requests on the given segment to the kernel
+(low level function).
+@param acquire_mutex specifies whether to lock array mutex
+*/
+static
+void
+os_aio_dispatch_read_array_submit_low(bool acquire_mutex MY_ATTRIBUTE((unused)))
+{
+#if defined(LINUX_NATIVE_AIO)
+	os_aio_array_t*	array = os_aio_read_array;
+	ulint		total_submitted = 0;
+	if (!srv_use_native_aio) {
+		return;
+	}
+	if (acquire_mutex)
+		os_mutex_enter(array->mutex);
+	/* Submit aio requests buffered on all segments. */
+	for (ulint i = 0; i < array->n_segments; i++) {
+		int	count = array->count[i];
+		if (count > 0) {
+			ulint	iocb_index = i * array->n_slots
+					     / array->n_segments;
+			int	submitted = io_submit(
+					array->aio_ctx[i], count,
+					&(array->pending[iocb_index]));
+			if (submitted == count) {
+				total_submitted += submitted;
+			} else {
+				/* io_submit returns number of successfully
+				queued requests or -errno. */
+				if (submitted >= 0) {
+					ib_logf(IB_LOG_LEVEL_ERROR,
+						"Trying to sumbit %d aio "
+						"requests, io_submit() "
+						"returns %d",
+						count, submitted);
+				} else {
+					const char*	errmsg =
+						strerror(-submitted);
+					ib_logf(IB_LOG_LEVEL_ERROR,
```

Comment:
IB_LOG_LEVEL_FATAL (sic?). The ut_error below can still be kept for safety

Reply:
done